Sources/HTTP/Models/Client/HTTP+Client.swift

```diff
@@ -82,26 +60,52 @@ where
 
     public func respond(to request: Request) throws -> Response {
         try assertValid(request)
-        guard !stream.isClosed else { throw ClientError.unableToConnect }
+        guard !stream.isClosed else {
+            throw ClientError.unableToConnect
+        }
+
+        ///  client MUST send a Host header field in all HTTP/1.1 request
+        /// messages.  If the target URI includes an authority component, then a
+        /// client MUST send a field-value for Host that is identical to that
+        /// authority component, excluding any userinfo subcomponent and its "@"
+        /// delimiter (Section 2.7.1).  If the authority component is missing or
+        /// undefined for the target URI, then a client MUST send a Host header
+        /// field with an empty field-value.
+        request.headers["Host"] = stream.hostname
+        request.headers["User-Agent"] = userAgent
+
+        let buffer = StreamBuffer<StreamType>(stream)
+        let serializer = Serializer(stream: buffer)
+        try serializer.serialize(request)
+
+        let parser = Parser(stream: buffer)
+        let response = try parser.parse()
+
+        response.peerAddress = parser.parsePeerAddress(
+            from: stream,
+            with: response.headers
+        )
+
+        try buffer.flush()
 
-        return try responder.respond(to: request)
+        return response
     }
 }
 
 let VERSION = "2"
 public var userAgent = "App (Swift) VaporEngine/\(VERSION)"
 
 
-extension ClientProtocol {
+extension Client {
     internal func assertValid(_ request: Request) throws {
-        if request.uri.host.isEmpty {
-            guard request.uri.host == host else {
+        if request.uri.hostname.isEmpty {
+            guard request.uri.hostname == hostname else {
                 throw ClientError.invalidRequestHost
             }
         }
 
         if request.uri.scheme.isEmpty {
-            guard request.uri.scheme.securityLayer.isSecure == securityLayer.isSecure else {
+            guard request.uri.scheme == scheme else {
```

Comment:
I think this might fail with websockets, technically `wss` and `https` are ok and both secure, but it's only under the isEmpty, so not sure.